src/controllers/sessionLockController.ts

```diff
@@ -243,6 +256,54 @@ export default class SessionLockController {
       )
     );
 
+  /**
+   * Lock a user authentication and clear all its session data
+   *
+   * @param req expects fiscal_code as a path param
+   *
+   * @returns a promise with the encoded response object
+   */
+  public readonly unlockUserAuthentication = (
+    req: express.Request
+  ): Promise<
+    IResponseErrorInternal | IResponseErrorValidation | IResponseNoContent
+  > =>
+    withFiscalCodeFromRequestParams(req, (fiscalCode) =>
+      withAuthUnlockBodyParams(req, (authUnlockBody) =>
+        pipe(
+          authUnlockBody.unlock_code,
+          O.fromNullable,
+          TE.of,
+          TE.bindTo("maybeUnlockCode"),
+          TE.bind("authLockData", () =>
+            this.getAuthLockDataOrError(fiscalCode)
+          ),
+          TE.filterOrElse(
+            ({ maybeUnlockCode, authLockData }) =>
+              O.isNone(maybeUnlockCode) ||
```

Comment:
I think we should return a `403` when the unlockCode provided by the user is wrong. see `403` comment in the spec

Reply:
Done in [0e1f892](https://github.com/pagopa/io-backend/pull/1065/commits/0e1f89265a2d893f2c10836e360bc1684ebf66d1), along with a query refactor